SimPPS/RPDigiProducer/plugins/RPGaussianTailNoiseAdder.cc

```diff
@@ -0,0 +1,46 @@
+#include "SimPPS/RPDigiProducer/plugins/RPGaussianTailNoiseAdder.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <iostream>
+#include "TMath.h"
+#include "TRandom.h"
+#include "CLHEP/Random/RandGauss.h"
+
+using namespace std;
+
+RPGaussianTailNoiseAdder::RPGaussianTailNoiseAdder(int numStrips,
+                                                   double theNoiseInElectrons,
+                                                   double theStripThresholdInE,
+                                                   int verbosity)
+    : numStrips_(numStrips), theNoiseInElectrons(theNoiseInElectrons), theStripThresholdInE(theStripThresholdInE) {
+  verbosity_ = verbosity;
+  strips_above_threshold_prob_ = TMath::Erfc(theStripThresholdInE / sqrt(2.0) / theNoiseInElectrons) / 2;
+}
+
+simromanpot::strip_charge_map RPGaussianTailNoiseAdder::addNoise(const simromanpot::strip_charge_map &theSignal) {
+  simromanpot::strip_charge_map the_strip_charge_map;
+
+  // noise on strips with signal:
+  for (simromanpot::strip_charge_map::const_iterator i = theSignal.begin(); i != theSignal.end(); ++i) {
+    double noise = CLHEP::RandGauss::shoot(0.0, theNoiseInElectrons);
```

Comment:
@mundim , normally CLHEP random engine should be defined in a producer and should be given by reference to all classes which sample digi. It is also possible if CLHEP random engine is initialized in the producer and use in this way (as it is done in hit production). Can you guarantee that the engine is properly initialized in the produce(...) method?

Reply:
Changes the constructor, passing the CLHEP::RandomEngine instanciated and inicialized by the producer.